primefaces/src/main/java/org/primefaces/util/FileUploadUtils.java

```diff
@@ -252,8 +252,25 @@ private static boolean isValidFileContent(PrimeApplicationContext context, FileU
                 }
             }
 
+
             String contentType = context.getFileTypeDetector().probeContentType(tempFile);
 
+            // We want to force the FileTypeDetector impl to try to detect the file type by its content,
+            // this is why we intentionally hide the original file name/extension at first because both
+            // tika and mime-types go the easy way of looking up the file extension if possible.
+            // If this first attempt failed we try again, but now while preserving the original file name/extension
+            // to e.g. make the JDK default FileTypeDetector work
+            if (contentType == null) {
+                String fileExtension = FilenameUtils.getExtension(fileName);
+
+                if (!fileExtension.isEmpty()) {
+                    String newFileName = FilenameUtils.removeExtension(tempFile.getFileName().toString()) + "." + fileExtension;
```

Comment:
tempFile is supposed to be without extension no?

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Files.html#createTempFile(java.lang.String,java.lang.String,java.nio.file.attribute.FileAttribute...)

> suffix - the suffix string to be used in generating the file's name; may be null, in which case ".tmp" is used

Maybe passing `""` as the suffix could work

Reply:
Yeah that seems to work, new code is pushed